File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model
{
    public abstract class Node : INode
    {
        public TextPosition Position
        {
            get;
            private set;
        }

        protected Node(TextPosition position)
        {
            Position = position;
        }

        public abstract void Accept(IQLVisitor visitor);

        public abstract T Accept<T>(IQLVisitor<T> visitor);
    }
}

File: JeffSebastian/playground/src/main/java/nl/uva/se/ql/ast/Node.java
package nl.uva.se.ql.ast;

public abstract class Node {
	
	private final int lineNumber;
	private final int offset;
	
	public Node(int lineNumber, int offset) {
		this.lineNumber = lineNumber;
		this.offset = offset;
	}

	public int getLineNumber() {
		return lineNumber;
	}

	public int getOffset() {
		return offset;
	}
	
}
